### ISSUE TITLE
Facetgrid: colors beyond range (extend) not saturated

### ISSUE THREAD

**Issue by lvankampenhout** [pydata/xarray#2932]
#### Code Sample, a copy-pastable example if possible

![Screen Shot 2019-04-30 at 11 59 30](https://user-images.githubusercontent.com/7933853/56954588-70238b00-6b3f-11e9-90ad-81045e500987.png)

Minimal example here:
https://github.com/lvankampenhout/bug-reports/blob/master/Facetgrid_cmap_extend.ipynb

#### Problem description

The extreme colors of neither the pcolormesh or colorbar (using extend='both') are not saturated as they should when faceting.

#### Output of ``xr.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.8 |Anaconda custom (x86_64)| (default, Dec 29 2018, 19:04:46) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
python-bits: 64
OS: Darwin
OS-release: 17.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: C
LANG: None
LOCALE: None.None
libhdf5: 1.10.1
libnetcdf: 4.4.1.1

xarray: 0.12.1
pandas: 0.23.4
numpy: 1.14.2
scipy: 0.18.1
netCDF4: 1.3.1
pydap: None
h5netcdf: None
h5py: 2.7.1
Nio: None
zarr: None
cftime: 1.0.0b1
nc_time_axis: None
PseudonetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: 1.2.0
dask: 0.13.0
distributed: None
matplotlib: 3.0.2
cartopy: 0.16.0
seaborn: 0.7.1
setuptools: 38.5.1
pip: 9.0.1
conda: 4.6.14
pytest: 3.0.5
IPython: 5.1.0
sphinx: 1.5.1
</details>

**Comment by mathause** [comment]
For completeness I add the minimal example

``` python
import numpy as np; import xarray as xr
da = xr.tutorial.open_dataset('air_temperature').air
cmap = 'RdBu_r'
levels = np.arange(240,300+1,10)
da.isel(time=slice(4)).plot(cmap=cmap, col='time', levels=levels, extend='both')
```

What I think happens is that the `cmap` is determined twice. First in [facetgrid.py](https://github.com/pydata/xarray/blob/master/xarray/plot/facetgrid.py#L225-L226). This returns a `matplotlib.colors.ListedColormap` with 6 colors. Then this reduced colormap is used in [plot.py](https://github.com/pydata/xarray/blob/master/xarray/plot/plot.py#L630-L631). However, now it only has 6 colors to choose from, so the `_over` and `_under` colors are the same as the first and last color.

**Comment by mathause** [comment]
The easiest way to fix this is probably to reset the cmap in [facetgrid.py Lines 224-227](https://github.com/pydata/xarray/blob/master/xarray/plot/facetgrid.py#L224-L227)
 ``` python

cmap = kwargs.get('cmap', None)

cmap_params, cbar_kwargs = _process_cmap_cbar_kwargs(func, kwargs, self.data.values)

 # reset the colormap #GH2932
 cmap_params['cmap'] = cmap
```

**Comment by lvankampenhout** [comment]
Thanks, I've implemented your suggestion as a workaround, but it fails with the following error:

`NameError: name '_process_cmap_cbar_kwargs' is not defined`

**Comment by dcherian** [comment]
See #3038

**Comment by dcherian** [comment]
Seems to work now? probably fixed by #3601

![7cfd12224bfde93bb1a8192bb4e9c2a82ab6f9a5](https://user-images.githubusercontent.com/2448579/77850265-d2a66080-71c0-11ea-840e-b5a8759ce12f.png)
